FHEM/lib/SD_Protocols.pm

```diff
@@ -1743,6 +1743,41 @@ sub _xFSK_methods_behind_here {
   # only for functionslist - no function!
 }
 
+=item ConvBresser_5in1()
+
+This function checks number/count of set bits within bytes 14-25 and inverted data of 13 byte further.
+Delete inverted data (nibble 1-27)and reduce message length (nibble 53).
+
+Input:  $hexData
+Output: $hexData
+        scalar converted message on success 
+        or array (1,"Error message")
+
+=cut
+
+sub ConvBresser_5in1 {
+  my $self    = shift // carp 'Not called within an object';
+  my $hexData = shift // croak 'Error: called without $hexdata as input';
+  my $d2;
+  my $bit;
+  my $bitsumRef;
+  my $bitadd = 0;
+  for (my $i = 0; $i < 13; $i++) {
+    $d2 = hex(substr($hexData,($i+13)*2,2));
+    return ( 1, qq[ConvBresser_5in1, inverted data at pos $i] ) if ((hex(substr($hexData,$i*2,2)) ^ $d2) != 255);
+    if ($i == 0) {
+      $bitsumRef = $d2;
+    }	else {
+      $bit = sprintf("%08b", $d2);
+      for (my $j = 0; $j < 8; $j++) {
+        $bitadd += substr($bit,$j,1);
```

Comment:
verstehe ich das richtig, dass hier die quersumme der gesetzte Bits beechnet wird und wir den Dezimalwert in einen String der die einzelnen Bits darstellt dafür umwandeln um anschließend wieder eine "Zahl" daraus zu machen?

Reply:
Hast du eine einfachere Lösung parat?

Reply:
mittels binär "und" einen counter hochzählen und das nächste bit nach links schieben.
Sobald $d2 0 also unwahr entspricht kann kein Bit mehr gesetzt sein und der Vorgang kann beendet werden.

In dem Beispiel fehlt leider die Perl Vairante.
https://www.geeksforgeeks.org/count-set-bits-in-an-integer/

Folgendes gerade durch den unitttest gejagt und funktioniert auch wie erwartet :)
```
      while ($d2) {
        $bitadd += $d2 & 1;
        $d2 >>= 1;
      }
```

Da $d2 auch nicht unendlich groß ist, besteht keine Gefahr einer Endlosschleife.

Reply:
Es müsste eigentlich auch mitels unpack gehen.  Das ist ja ein schweizer Taschenmesser.